src/main/java/org/dspace/ctask/replicate/AbstractPackagerTask.java

```diff
@@ -75,32 +74,29 @@ protected PackageParameters loadPackagerParameters(String moduleName)
             //loop through all properties in the config file
             for(String property : moduleProps.stringPropertyNames())
             {
-                //Only obey the setting(s) beginning with this task's ID/name, 
-                if(property.startsWith(this.taskId))
-                {
-                    //Parse out the option name by removing the "[taskID]." from beginning of property
-                    String option = property.replace(taskId + ".", "");
-                    String value = moduleProps.getProperty(property);
-                    
-                    //Check which option is being set
-                    if(option.equalsIgnoreCase(recursiveMode))
-                    {
-                        pkgParams.setRecursiveModeEnabled(Boolean.parseBoolean(value));
-                    }
-                    else if (option.equals(useWorkflow))
-                    {
-                        pkgParams.setWorkflowEnabled(Boolean.parseBoolean(value));
-                    } 
-                    else if (option.equals(useCollectionTemplate))
-                    {
-                        pkgParams.setUseCollectionTemplate(Boolean.parseBoolean(value));
-                    }
-                    else //otherwise, assume the Packager will understand what to do with this option
-                    {
-                        //just set it as a property in PackageParameters
-                        pkgParams.addProperty(option, value);
+                // Only obey the setting(s) belonging to the requested module
+                if (property.startsWith(moduleName)) {
```

Comment:
Rather than looping through all properties and then checking if they start with a specific prefix (`moduleName`), you can now simply use `configurationService.getPropertyKeys(moduleName)` to select all the configs that start with that `moduleName`.

https://github.com/DSpace/DSpace/blob/master/dspace-services/src/main/java/org/dspace/services/ConfigurationService.java#L109